docs/src/operations/authorization.md

```diff
@@ -0,0 +1,63 @@
+# Authorization
+
+Zeebe clients also provide a way for users to modify gRPC call headers, namely to provide authorization credentials. Note that the gateway doesn't provide any way of validating these headers, so users must implement a reverse proxy with a gRPC interceptor to validate the credentials.
+
+## Java
+
+Users can modify gRPC headers by providing a custom [CredentialsProvider](https://github.com/zeebe-io/zeebe/blob/develop/clients/java/src/main/java/io/zeebe/client/CredentialsProvider.java). The CredentialsProvider interface consists of a single `applyCredentials(Metadata headers)` method which takes a map of headers to which it should add credentials:
+
+```java
+public class MyCredentialsProvider implements CredentialsProvider {
+  /**
+   * Adds a JSON Web Token (JWT) to the Authorization header of a gRPC call. The JWT is obtained from a provided Zeebe credentials YAML file.
+   */
+  @Override
+  public void applyCredentials(Metadata headers) {
+    final Key<String> authHeaderkey = Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);     
+    headers.put(authHeaderKey, "Bearer someKindOfToken");
+  }
+}
+``` 
+
+After implementing the CredentialsProvider, we can simply provide it when building a client:
+
+```java
+public class SecureClient {
+
+  public static void main(final String[] args) {
+    final ZeebeClient client = ZeebeClient.newClientBuilder().credentialsProvider(new MyCredentialsProvider()).build();
+
+    // continue...    
+  }
+}
+```
+
+It should be noted that the CredentialsProvider interface is unstable and will probably be augmented in the near future, as Zeebe's authorization features are improved. For instance, if the validating interceptor rejects the request with an UNAUTHENTICATED status code, the request will just fail. In the future, we'd like to improve this behavior to deal with UNAUTHENTICATED failures by refreshing the credentials and, if new credentials are present, retrying the request. Doing this relieves users from having check expiration times at every request and also provides more robust error handling. However, this will also require changes to the CredentialsProvider interface.
```

Comment:
Don't get people's hopes up :stuck_out_tongue: 

Let's just write that it's still in development (unstable or experimental whichever), but let's not mention future concepts.

Reply:
True :smile: I may have gotten ahead of myself there